### PR TITLE
[MAINTENANCE]: Refactor ExpectationSuite to include profiler_config in citations

### PR DESCRIPTION
**Pull request by cdkini**
Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE],  [DOCS], or [MAINTENANCE]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html).

Changes proposed in this pull request:
- Add `profiler_config` to `add_citation` and `get_citations` methods of `ExpectationSuite`
- Add unit tests to cover above changes

After submitting your PR, CI checks will run and @tiny-tim-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/en/latest/contributing/style_guide.html?highlight=style%20guide)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/en/latest/contributing/contribution_checklist.html?highlight=checklist) of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/en/latest/contributing/testing.html#contributing-testing-writing-unit-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/core/test_expectation_suite.py [hunk 1]
```diff
@@ -309,6 +309,21 @@ def test_add_citation(baseline_suite):
     assert baseline_suite.meta["citations"][0].get("comment") == "hello!"
 
 
+def test_add_citation_with_profiler_config(baseline_suite):
```
Comment (cdkini) [inline tests/core/test_expectation_suite.py]: Added some basic tests here but let me know if we need to add anything additional to gain coverage over the changes. 

I used `{"variables": {}, "rules": {}}` as the `profiler_config` for all tests; I didn't think we needed a fixture but let me know if you think otherwise.

**Review by cdkini — COMMENTED** [review]
Left basic clarifying questions as self-review.

**Review by alexsherstinsky — DISMISSED** [review]
@cdkini do we want to add a test in which the contents of `profile_config` are not so empty?..  It might be good to insure that the full dictionary is properly added to a citation.

**Comment by cdkini** [comment]
@alexsherstinsky I think that's a fair critique and a simple enough change. Do we want to go to the full extent of a mock config akin to something like `alice` or `bob` or do you think it would be okay to just hardcode values?

**Comment by alexsherstinsky** [comment]
> @alexsherstinsky I think that's a fair critique and a simple enough change. Do we want to go to the full extent of a mock config akin to something like `alice` or `bob` or do you think it would be okay to just hardcode values?

@cdkini In my opinion a simpler hardcoding would be fine -- personally, whichever is easier for you (though `bobster` is pretty simple, or `alice`, etc.).  What does @talagluck think?

**Review by talagluck — APPROVED** [review]
Looks great!
